x/emissions/keeper/queryserver/query_server_stake.go

```diff
@@ -114,3 +115,80 @@ func (qs queryServer) GetTopicStake(ctx context.Context, req *types.QueryTopicSt
 
 	return &types.QueryTopicStakeResponse{Amount: stake}, nil
 }
+
+func (qs queryServer) GetStakeRemovalsForBlock(
+	ctx context.Context,
+	req *types.QueryStakeRemovalsForBlockRequest,
+) (*types.QueryStakeRemovalsForBlockResponse, error) {
+	removals, err := qs.k.GetStakeRemovalsForBlock(ctx, req.BlockHeight)
+	if err != nil {
+		return nil, err
+	}
+	maxLimit := types.DefaultParams().MaxPageLimit
+	moduleParams, err := qs.k.GetParams(ctx)
+	if err == nil {
+		maxLimit = moduleParams.MaxPageLimit
+	}
+	if uint64(len(removals)) > maxLimit {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("cannot query more than %d removals at once", maxLimit))
+	}
+	removalPointers := make([]*types.StakeRemovalInfo, len(removals))
+	for i, removal := range removals {
+		removalPointers[i] = &removal
+	}
+	return &types.QueryStakeRemovalsForBlockResponse{Removals: removalPointers}, nil
+}
+
+func (qs queryServer) GetDelegateStakeRemovalsForBlock(
+	ctx context.Context,
+	req *types.QueryDelegateStakeRemovalsForBlockRequest,
+) (*types.QueryDelegateStakeRemovalsForBlockResponse, error) {
+	removals, err := qs.k.GetDelegateStakeRemovalsForBlock(ctx, req.BlockHeight)
+	if err != nil {
+		return nil, err
+	}
+	maxLimit := types.DefaultParams().MaxPageLimit
+	moduleParams, err := qs.k.GetParams(ctx)
+	if err == nil {
+		maxLimit = moduleParams.MaxPageLimit
+	}
+	if uint64(len(removals)) > maxLimit {
+		return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("cannot query more than %d removals at once", maxLimit))
```

Comment:
can't you just return the ones that are less than the page limit? Or if there are too many that means you can't get them at all?

Reply:
this is just copied and pasted from how our other "pagination" query server functions work

but yeah the right thing to do would not be to deny the user at least some portion of their data

Reply:
i'll fix it here but keep in mind this is existing elsewhere in our query server functions